src/main/java/pro/cloudnode/smp/bankaccounts/events/PlayerInteract.java

```diff
@@ -24,34 +27,54 @@ public void onPlayerInteractEvent(PlayerInteractEvent event) {
         final @NotNull Player player = event.getPlayer();
         if (event.getAction() == Action.RIGHT_CLICK_BLOCK) {
             final @NotNull Optional<Block> block = Optional.ofNullable(event.getClickedBlock());
-            if (block.isPresent() && block.get().getState() instanceof final @NotNull Chest chest && !chest.getInventory().isEmpty()) {
-                final @NotNull Optional<POS> pos = POS.get(block.get());
-                if (pos.isEmpty()) return;
-                event.setCancelled(true);
-                if (player.getUniqueId().equals(pos.get().seller.owner.getUniqueId())) {
-                    POS.openOwnerGui(player, chest, pos.get());
-                    return;
-                }
-                if (!player.hasPermission("bank.pos.use")) {
-                    player.sendMessage(MiniMessage.miniMessage().deserialize(Objects.requireNonNull(BankAccounts.getInstance().getConfig().getString("messages.errors.pos-no-permission"))));
-                    return;
+            Optional<POS> pos = Optional.empty();
+
+            if (block.isEmpty()) return;
+
+            // handle double chests by checking if either side is a POS
+            if (block.get().getState() instanceof Chest doubleChest && doubleChest.getInventory() instanceof DoubleChestInventory) {
+                final @NotNull Chest leftChest = (Chest) Objects.requireNonNull(((DoubleChestInventory) doubleChest.getInventory()).getLeftSide().getHolder());
+                final @NotNull Chest rightChest = (Chest) Objects.requireNonNull(((DoubleChestInventory) doubleChest.getInventory()).getRightSide().getHolder());
+
+                if (POS.get(leftChest).isPresent()) {
+                    pos = POS.get(leftChest);
+                } else if (POS.get(rightChest).isPresent()) {
+                    pos = POS.get(rightChest);
                 }
-                final @NotNull ItemStack heldItem = player.getInventory().getItemInMainHand();
-                if (heldItem.getType() == Material.AIR) {
-                    player.sendMessage(MiniMessage.miniMessage().deserialize(Objects.requireNonNull(BankAccounts.getInstance().getConfig().getString("messages.errors.no-card"))));
+            } else if (block.get().getState() instanceof final @NotNull Chest chest && !chest.getInventory().isEmpty()) {
+                // handle single chests
+                pos = POS.get(block.get());
+            }
+
+            if (pos.isEmpty()) return;
+            event.setCancelled(true);
+
+            Chest chest = (Chest) block.get().getState(); // the handling of double chests is done inside POS#get
```

Comment:
```suggestion
            final @NotNull Chest chest = (Chest) block.get().getState(); // the handling of double chests is done inside POS#get
```

wouldnt it be good here to do an instanceof check?

Reply:
well we can just return if its not a double chest nor a normal chest